devdoc/watchdog.md

```diff
@@ -59,7 +60,10 @@ Callback for the timer.
 
 **SRS_WATCHDOG_42_027: [** If `context` is `NULL` then `watchdog_expired_callback` shall terminate the process. **]**
 
-**SRS_WATCHDOG_42_021: [** If the state of the watchdog is `RUNNING` then `watchdog_expired_callback` shall call `callback` with the `context` and `message` from `watchdog_start`. **]**
+**SRS_WATCHDOG_45_005: [** If the state of the watchdog is `RUNNING` then **]**
+- **SRS_WATCHDOG_45_001: [** `watchdog_expired_callback` shall set the state to `EXPIRING` **]**
+- **SRS_WATCHDOG_42_021: [** `watchdog_expired_callback` shall call `callback` with the `context` and `message` from `watchdog_start`. **]**
+- **SRS_WATCHDOG_45_002: [** `watchdog_expired_callback` shall return the state to `RUNNING`. **]**

```

Comment:
>e to `RUNNING`.  [](http://example.com/codeflow?start=78&length=16)

This implies that we're back to the same state - in which we could have the watchdog_expired_callback called and all would be well - we would execute again the user callback for the watchdog.

Shouldn't this user callback only be executed once?

Reply:
In my mind it sounds like the state machine should have a separate state like USER_CALLBACK_CALLED

Reply:
Switched to state management for... state management.